lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,755 @@
+so you can get all the data you need from a response:
+
+  $t->put_ok('/bees', json => { type => 'worker', name => 'Karl' })
+    ->status_is(202)
+    ->json_has('/id');
+
+  # Pull out the id from the response
+  my $newbee = $t->tx->res->json('/id');
+
+  # Make a new request with data from the previous response
+  $t->get_ok("/bees/$newbee")
+    ->status_is(200)
+    ->json_is('/name' => 'Karl');
+
+The L<Test::Mojo> object is I<stateful>. As long as we haven't started a new
+transaction by invoking one of the C<*_ok> methods, the request and response
+objects from the previous transaction are available in the L<Test::Mojo>
+object:
+
+  # First transaction
+  $t->get_ok('/frogs?q=bullfrog', { 'Content-Type' => 'application/json' })
+    ->status_is(200)
+    ->json_like('/0/species' => qr/catesbeianus/i);
+
+  # Still first transaction
+  $t->content_type_is('application/json');
+
+  # Second transaction; Test::Mojo tx object is reset
+  $t->get_ok('/frogs?q=banjo', { 'Content-Type' => 'text/html' })
+    ->status_is(200)
+    ->content_like(qr/interioris/i);
+
+  # Second transaction again
+  $t->content_type_is('text/html');
+
+This statefulness also enables L<Test::Mojo> to handle sessions, follow
+redirects, and inspect past responses during a redirect.
+
+=head2 The L<Test::Mojo> Object
+
+The L<Test::Mojo> object manages the Mojolicious application lifecycle (if a
+Mojolicious application class is supplied) as well as exposes the built-in
+L<Mojo::UserAgent> object. To create a bare L<Test::Mojo> object:
+
+  my $t = Test::Mojo->new;
+
+This object initializes a L<Mojo::UserAgent> object and provides a variety of
+test assertion methods for accessing a web application. For example, with this
+object, we could test any running web application:
+
+  $t->get_ok('https://www.google.com/')
+    ->status_is(200)
+    ->content_like(qr/search/i);
+
+You can access the user agent directly if you want to make web requests
+without triggering test assertions:
+
+  my $tx = $t->ua->post('https://duckduckgo.com/html',
+                        form => {q => 'hypnotoad'});
+
+  $tx->result->dom->find('a.result__a')
+    ->each(sub { say $_->text });
+
+See L<Mojo::UserAgent> for the complete API and return values.
+
+=head2 Testing Mojolicious Applications
+
+If you pass the name of a L<Mojolicious> application class (e.g., 'MyApp') to
+the L<Test::Mojo> constructor, L<Test::Mojo> will instantiate the class and
+start it, and cause it to listen on a random (unused) port number. Testing a
+Mojolicious application using L<Test::Mojo> will never conflict with running
+applications, including the application you're testing.
+
+The L<Mojo::UserAgent> object in L<Test::Mojo> will know where the application
+is running and make requests to it. Once the tests have completed, the
+L<Mojolicious> application will be torn down.
+
+  my $t = Test::Mojo->new('Frogs');  ## runs on localhost:32114
```

Comment:
`...;  ## runs on localhost:32114` is not really a comment style we use in Mojolicious.